### ISSUE TITLE
Expo authentication using next-auth

### ISSUE THREAD

**Issue by rahmat-dev** [t3-oss/create-t3-turbo#486]
Can I use the existing next-auth configuration to handle authentication for the expo app? I don't wanna use any service like clerk, supabase, etc.

**Comment by philsmithies** [comment]
right now it is just with external services it seems

**Comment by BRDSLY** [comment]
I believe we're still waiting on authjs/next. Keep an eye on these PRs.

https://github.com/nextauthjs/next-auth/pull/5240
https://github.com/t3-oss/create-t3-turbo/pull/133

**Comment by HarunKilic** [comment]
Is there any workaround except external services?

**Comment by BRDSLY** [comment]
In one of my projects I did a little bit of funkiness but not sure on the security etc. Can't remember exactly how I did it but something along the following:

- Used expo oauth to get all details of google login, send token and other details to a custom mobile login endpoint
- Generate unique device ID stored in expo securestore
- check the provided token using google api (This is good enough for me to allow them access), make real account if not an existing user
- Manually created a new session I think, and a new mobile session object that contains that session, unique device ID that will be bound to that session and provided on every call I think?
- added to the default nextauth prisma adapter to do session logic, think I maybe added some middleware in to facilitate there being now 2 auth objects

Haven't touched this project in a while, can't remember it well, but have a play around with your next auth adapter which will be a drizzle one now I think? haven't used t3 in a while so things will probably have changed.
